app/models/ability.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    
+    if user.admin?
+      can :manage, [User, Group], company_id: user.company_id
+      cannot [:destroy, :edit], user
+    elsif user.supervisor
+      can :read, [User, Group], company_id: user.company_id
+    elsif user.member
+      can :read, [User, Group], company_id: user.company_id
```

Comment:
member should not read users and groups 
member can only fill surveys which they are allowed to